Non-Interactive-Protocol.md

```diff
@@ -0,0 +1,175 @@
+transaction paired with the outcome.
+
+2. The oracle doesn't release the outcome signature, independently of counterparty
+behavior, the party must broadcast the refund transaction after its timelock expiration.
+
+# Table of Contents
+	* [General Requirements](#general-requirements)
+	* [Fee-Bumping Policy](#fee-bumping-policy)
+	* [Funding Phase](#funding-phase)
+	* [Execution Phase](#execution-phase)
+	* [Reorgs](#reorgs)
+
+# General Requirements
+
+A DLC client should have censorship-resistant access to the blockchain, use a
+local fee-estimator and ensure it's well-connected to the tx-relay network.
+
+We recommend to not react on any mempool event as otherwise it would be a source of
+cross-layer mapping.
+
+We recommend to not sign transactions until their broadcast is required
+to avoid their misuage in case of site compromise, assuming an external signer.
+
+# Fee bumping policy
+
+Client's `congestion_delay`/`security_delay` can be scheduled on three different basis:
+- a block height
+- a local clock
+- an observed feerate fluctuation
+
+We don't provide recommendations for the congestion delay as its settings depends of user
+liquidity preferences and available local data to predict the feerate.
+
+As `security_delay` must be offensive, we recommend to set it to 1 block.
+
+
+# Funding Phase
+
+A party always starts the protocol in the funding phase. The funding phase can
+terminate by either a transition to the execution phase or a protocol abort.
+
+## Requirements
+
+A node:
+	- if a counterparty's funding signature is received:
+		- MUST finalize the funding transaction with a local `SIGHASH_ALL` signature
+		- MUST broadcast the funding transaction
+
+A node:
+	- if a funding signature has been sent or a funding transaction broadcast and there is no confirmation after 6 blocks:
+		- MUST spend the change output with a CPFP increasing package feerate to shorten confirmation time
+		- MUST start a bumping timer of length `congestion_delay`
+		- if `congestion_delay` expires:
+			- MUST spend the change output with a CPFP increasing package feerate to shorten confirmation time
+			- MUST replace the CPFP transaction according to BIP 125
+			- MUST reschedule timer for next bump
+		- if `contract_maturity_bound` is 20 blocks ahead from local tip:
+			- MUST spend the funding input with a higher-feerate transaction/absolute fee than any previously propagated package feerate
+			- MUST start a bumping timer of length `security_delay`
+			- MUST disable `congestion_delay` timer
+		- if `security_delay` expires:
+			- MUST spend the funding input with a higher-feerate transaction/absolute fee than any previously propagated package feerate
+			- MUST replace the CPFP transaction according to BIP 125
+			- MUST reschedule timer for next bump
+		- MUST NOT spend more in bumping fees than its collateral input value
+
+A node:
+	- if the funding transaction confirms:
+		- MUST track spend of the CET output
```

Comment:
isn't it the funding output they need to track? Perhaps that's what you mean and are using different terminology? (CET output would be the final payout whereas funding output is the output spent by CETs)

Reply:
You're right, my mistake.

Reply:
I would say there is a funding tx output spent by a CET, and I wouldn't attach ownership of the redeem script to CETs as they all share so that seems to be in the domain of the funding transaction/funding output still. But I think regardless of this minute difference the wording in the spec is a little confusing as the thing a client must track to see if it is spent is the funding output not the CET (just like in LN you track the funding output)

Reply:
Good to me, updated.